doc/neps/nep-0019-rng-policy.rst

```diff
@@ -0,0 +1,208 @@
+
+We also had not thought very thoroughly about the limits of what we really
+could promise (and by “we” in this section, we really mean Robert Kern, let’s
+be honest).  Despite all of the weasel words, our policy overpromises
+compatibility.  The same version of ``numpy`` built on different platforms, or
+just in a different way could cause changes in the stream, with varying degrees
+of rarity.  The biggest is that the ``.multivariate_normal()`` method relies on
+``numpy.linalg`` functions.  Even on the same platform, if one links ``numpy``
+with a different LAPACK, ``.multivariate_normal()`` may well return completely
+different results.  More rarely, building on a different OS or CPU can cause
+differences in the stream.  We use C ``long`` integers internally for integer
+distribution (it seemed like a good idea at the time), and those can vary in
+size depending on the platform.  Distribution methods can overflow their
+internal C ``longs`` at different breakpoints depending on the platform and
+cause all of the random variate draws that follow to be different.
+
+And even if all of that is controlled, our policy still does not provide exact
+guarantees across versions.  We still do apply bug fixes when correctness is at
+stake.  And even if we didn’t do that, any nontrivial program does more than
+just draw random numbers.  They do computations on those numbers, transform
+those with numerical algorithms from the rest of ``numpy``, which is not
+subject to so strict a policy.  Trying to maintain stream-compatibility for our
+random number distributions does not help reproducible research for these
+reasons.
+
+The standard practice now for bit-for-bit reproducible research is to pin all
+of the versions of code of your software stack, possibly down to the OS itself.
+The landscape for accomplishing this is much easier today than it was in 2008.
+We now have ``pip``.  We now have virtual machines.  Those who need to
+reproduce simulations exactly now can (and ought to) do so by using the exact
+same version of ``numpy``.  We do not need to maintain stream-compatibility
+across ``numpy`` versions to help them.
+
+Our stream-compatibility guarantee has hindered our ability to make
+improvements to ``numpy.random``.  Several first-time contributors have
+submitted PRs to improve the distributions, usually by implementing a faster,
+or more accurate algorithm than the one that is currently there.
+Unfortunately, most of them would have required breaking the stream to do so.
+Blocked by our policy, and our inability to work around that policy, many of
+those contributors simply walked away.
+
+
+Implementation
+--------------
+
+We propose first freezing ``RandomState`` as it is and developing a new RNG
+subsystem alongside it.  This allows anyone who has been relying on our old
+stream-compatibility guarantee to have plenty of time to migrate.
+``RandomState`` will be considered deprecated, but with a long deprecation
```

Comment:
RandomState could pretty easily be spun out into a stand-alone package, if useful. It is effectively a stand-alone submodule already.